assets/src/setup/components/selectable/index.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import './style.css';
+
+/**
+ * A component providing different visual states depending on whether the element is selectable.
+ *
+ * @param {Object} props Component props.
+ * @param {any} props.children Component children.
+ * @param {string} props.className Extra classes to add to the element.
+ * @param {string} props.direction The direction in which the visual treatment of the selectable element will orient.
+ * @param {string} props.HTMLElement The HTML element to serve as the selectable wrapper.
+ * @param {boolean} props.selected Whether the element is selected.
+ */
+export function Selectable( { children, className, direction, HTMLElement, selected, ...props } ) {
+	return (
+		<HTMLElement
+			className={
+				`${ className } selectable ${ selected ? 'selectable--selected' : '' } selectable--${ direction }`
+					.replace( /  /g, ' ' )
+					.trim()
```

Comment:
Is this needed? Would spaces between the classes alter the component in any way?

Reply:
Good question. My thinking with this was to make writing the unit tests easier, but on second thought there's no reason I can't just put the extra spaces into the unit tests. Then we can avoid doing this extra bit of work on this side. Updated.

Reply:
Oh OK. If no extra space between the classes is still desired, you could probably try something like:

```js
	const classNames = [ className, 'selectable', ( selected ? 'selectable--selected' : '' ), `selectable--${ direction }` ];

	return (
		<HTMLElement
			className={ classNames.filter( ( name ) => name ).join( ' ' ) }
```

Reply:
I like it. I made this change 😄